core/src/main/scala/org/apache/spark/status/AppStatusListener.scala

```diff
@@ -211,6 +211,31 @@ private[spark] class AppStatusListener(
           update(rdd, now)
         }
       }
+      // remove partition information
+      liveRDDs.values.foreach { rdd =>
+        rdd.getPartitions.values
+          .filter(_.executors.contains(event.executorId)).foreach { partition =>
+          partition.executors.filter(_ == event.executorId).foreach { _ =>
+            if (partition.executors.length == 1) {
+              rdd.removePartition(partition.blockName)
+              rdd.memoryUsed = addDeltaToValue(rdd.memoryUsed, partition.memoryUsed * -1)
+              rdd.diskUsed = addDeltaToValue(rdd.diskUsed, partition.diskUsed * -1)
+            } else {
+              rdd.memoryUsed = addDeltaToValue(rdd.memoryUsed,
+                (partition.memoryUsed / partition.executors.length) * -1)
+              rdd.diskUsed = addDeltaToValue(rdd.diskUsed,
+                (partition.diskUsed / partition.executors.length) * -1)
+              partition.update(partition.executors.filter(!_.equals(event.executorId)),
+                rdd.storageLevel,
+                addDeltaToValue(partition.memoryUsed,
+                  (partition.memoryUsed / partition.executors.length) * -1),
+                addDeltaToValue(partition.diskUsed,
+                  (partition.diskUsed / partition.executors.length) * -1))
+            }
+            update(rdd, now)
```

Comment:
You're calling this multiple times unnecessarily (one for each partition). Should be called only once if there have been any updates.

Reply:
moved update outside the partition loop